### PR TITLE
client: keyspace client

### PR DESCRIPTION
**Pull request by AmoebaProtozoa**
Added keyspace management gRPC client
Issue Number: Ref #5293 

Note: In order to run tests, this PR contains changes from https://github.com/tikv/pd/pull/5265 https://github.com/tikv/pd/pull/5291
New Changes:
Client: under `client/`
Integration tests: under `tests/client`

### What is changed and how does it work?

```commit-message
client: keyspace management gRPC client
```

### Check List

Tests

- Integration Test

Code changes

### Release note

```release-note
None
```

### CODE REVIEW

File: client/keyspace_client.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"github.com/opentracing/opentracing-go"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/keyspacepb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/client/grpcutil"
+	"google.golang.org/grpc"
+)
+
+// KeyspaceClient manages keyspace metadata.
+type KeyspaceClient interface {
```
Comment (ystaticy) [inline client/keyspace_client.go:31]: It's need more test case.

File: client/keyspace_client.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	// UpdateKeyspaceConfig updates target keyspace's config.
+	UpdateKeyspaceConfig(ctx context.Context, name string, put map[string]string, delete []string) (*keyspacepb.KeyspaceMeta, error)
```
Comment (ystaticy) [inline client/keyspace_client.go:33]: Which scenario probably call this interface?

**Comment by nolouch** [comment]
@AmoebaProtozoa  Could you plz resolve the conflicts?

**Review by disksing — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**